### ISSUE TITLE
Make 'notify_team_new_comment' GitHub action re-usable

### ISSUE THREAD

**Issue by MisRob** [learningequality/kolibri#12567]
## Summary

In our repositories, there are three same copies of the GitHub action that sends a Slack notification when a contributor comments on issue, [.github/workflows/notify_team_new_comment.yml](https://github.com/learningequality/kolibri/blob/develop/.github/workflows/notify_team_new_comment.yml). The other two repositories that have this action in their `.github/workflows` are Studio and Kolibri Design System.

This is cumbersome since we need three pull requests for each bugfix. This gets amplified by troubles with testing GH actions when we often need to try out more changes to apply a fix.

The goal of this issue is to find a place for this action in our GitHub organization and share it from there between Kolibri, Studio, and Kolibri Design System.

**Comment by MisRob** [comment]
I believe @rtibbles mentioned that [https://github.com/learningequality/.github](https://github.com/learningequality/.github) repository could be utilized

**Comment by rtibbles** [comment]
Yes, I think the setup would be that we make it as a reusable workflow in the `.github` repository and then update every place it is currently used to reference it.

**Comment by MisRob** [comment]
- For an example of a reusable `.github` action, see
  - https://github.com/learningequality/.github/blob/main/.github/workflows/update-pr-spreadsheet.yml
  - https://github.com/learningequality/test-actions/blob/main/.github/workflows/update-pr-spreadsheet.yml

**Comment by rparadowski** [comment]
I will make it if it's still free.

**Comment by akolson** [comment]
Hi @rparadowski  thanks you. I have assigned it to you!

**Comment by MisRob** [comment]
Hi @rparadowski, I'm reaching out based on your e-mail. Here's some guidance on how to best organize this work, so we can test it without breaking the production:

- (1) The 1st pull request with the re-usable shared logic to target https://github.com/learningequality/.github repository
- (2) The 2nd pull request with a small action that re-uses (1) to target https://github.com/learningequality/test-actions. Imagine `test-actions` as a temporary representative of Kolibri/Studio/KDS.

We would then review and merge both, and tested it works in `test-actions`. After that'd be done, we would proceed to move the part from `test-actions` to Kolibri/Studio/KDS. Let us know if you had any questions.

Before you open `.github` and `test-actions` pull requests, it'd be good to have your two own repositories for easier development. However, you may not be able to test it as a whole, and that's not a big deal - we will collaborate with you on fine-tuning the logic to make it work in LE repositories.

**Comment by rparadowski** [comment]
Hi, thank you for instruction.

Note for testing: default branch of dependent repo (eg kolibri, test-actions) needs to be switched to change branch of the PR in order to test the change (https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#issue_comment).

1. .github PR created: https://github.com/learningequality/.github/pull/9
2. test-actions PR created https://github.com/learningequality/test-actions/pull/53 and tested using https://github.com/rparadowski/test-actions/pull/1 and private slack. For tests I removed constraints on author and then reverted it.

**Comment by MisRob** [comment]
Hi @rparadowski, thanks a lot! Please give me few days to test and preview. I should get back to you no later than some time next week.

**Comment by MisRob** [comment]
Hi @rparadowski, I was just able to successfully test this, thanks so much. Would you be interested to complete this issue by opening PRs to all repos that will use `.github` shared workflow? Please let me know and I can provide more guidance.

**Comment by MisRob** [comment]
@rparadowski On `.github` part, the only tweak I noticed that would be needed is [here](https://github.com/learningequality/.github/pull/9/files#diff-56aa4481a772073713de8422284baf38d8775f6f9fc7609348750fdc0bdb896cR34)

```
text: "*[Kolibri] New comment on issue: <${{ github.event.issue.html_url }}#issuecomment-${{ github.event.comment.id }}|${{ steps.escape_title.outputs.ISSUE_TITLE }} by ${{ github.event.comment.user.login }}>*"
```

instead of `[Kolibri]` we need to provide repository name, because this will be different for each repo where we call the shared workflow.

**Comment by rparadowski** [comment]
@MisRob In case repo name doesn't have to be capitalized, here is dynamic repo name:
https://github.com/learningequality/.github/pull/11
![Image](https://github.com/user-attachments/assets/0c2256db-b8e4-4804-9afa-390b952c0274)

Use reusable workflow in other repos:
https://github.com/learningequality/kolibri/pull/12900
https://github.com/learningequality/studio/pull/4844
https://github.com/learningequality/kolibri-design-system/pull/849
https://github.com/learningequality/ricecooker/pull/543

**Comment by rparadowski** [comment]
@MisRob and also that one to fix early merge of https://github.com/learningequality/test-actions/pull/53

https://github.com/learningequality/test-actions/pull/55/files

**Comment by MisRob** [comment]
Thanks a lot @rparadowski, we will review. The repo name doesn't need to be capitalized, no problem at all.

**Comment by MisRob** [comment]
Hi @rparadowski, so far me merged https://github.com/learningequality/kolibri/pull/12900 and it seems to work well so I anticipate we will merge the remaining PRs soon. We use these automations from `le-utils` repo as well, if you'd like to open the last pull request, but no problem I can do it too. Thanks a lot for this work.

**Comment by rparadowski** [comment]
I have to admit I don't fully understand. Do you want to:
1. move `notify_team_new_comment` workflow definition from .github repo to le-utils repo?
2. Add additional workflow to le-utils repo so it starts using notify_team_new_comment from now on?

**Comment by MisRob** [comment]
H @rparadowski, just (2) :-) Having this https://github.com/rparadowski/kolibri/blob/cd9a6310eddc2e48f5eb739981cee531468c54d0/.github/workflows/notify_team_new_comment.yml in `le-utils`.

**Comment by MisRob** [comment]
Hi @rparadowski,  I wanted to mention that [Learning Equality will be closed from December 23 to January 5](https://github.com/learningequality/kolibri/discussions/12956).

**Comment by rparadowski** [comment]
Hi @MisRob and sorry for lack of responsiveness from my side. Please see requested PR: https://github.com/learningequality/le-utils/pull/151

**Comment by MisRob** [comment]
Thank you @rparadowski
